README.md

```diff
@@ -1,16 +1,644 @@
+│                                                           │         │           ▲    │         │        │
+│                                                           └─────────┘           ┆    └─────────┘        │
+│                                                                               Using                     │
+│                                                                               Result                    │
+│                                                                  ┌─────────┐   Of         ┌─────────┐   │
+│                                                                  │         │    ┆         │         │   │
+│                                                                  │   Dan   ╞════Set DNS══►│  Erin   │   │
+│                                                                  │         │              │         │   │
+│                                                                  └─────────┘              └─────────┘   │
+│                                                                                                         │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+# 2 Roles
+
+Invocation adds two new roles to UCAN: invoker and executor. The existing UCAN delegator and delegate principals MUST persist to the invocation.
+
+| UCAN Field | Delegation                             | Invocation                        |
+|------------|----------------------------------------|-----------------------------------|
+| `iss`      | Delegator: transfer authority (active) | Invoker: request action (active)  |
+| `aud`      | Delegate: gain authority (passive)     | Executor: perform action (active) |
+
+## 2.1 Invoker
+
+The invoker signals to the executor that a task associated with a UCAN SHOULD be performed.
+
+The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `iss` field of the contained UCAN.
+
+## 2.2 Executor
+
+The executor is directed to perform some task described in the UCAN by the invoker.
+
+The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
+
+# 3 Envelope
+
+The invocation envelope is a thin wrapper around a UCAN that conveys that all of the contained UCAN actions SHOULD be performed.
+
+All of the roles from the referenced UCANs MUST persist to the invocation per the [roles table](#2-roles).
+
+The invocation wrapper MUST be signed by the same principal that issued the UCAN.
+
+## 3.1 Fields
+
+| Field         | Type                                         | Value     | Description                                      | Required | Default |
+|---------------|----------------------------------------------|-----------|--------------------------------------------------|----------|---------|
+| `ucan/invoke` | `CID`                                        |           | The CID of the UCAN to invoke                    | Yes      |         |
```

Comment:
What's the reason for having `ucan` prefixed field name as opposed to regular one ?

Also is there reason to limit this to a single UCAN, e.g why not just have a same `prf` field from UCAN ?

Reply:
@QuinnWilton also mentioned that having it be more like `prf` would be what she expected, too. I had it initially, and in an attempt to simplify reduced it to one.

The `ucan/invoke` field name is currently doing double duty:

```js
{
  "type": "ucan/invoke",
  "prf": ["Qm12345"]
  // ...
}
```

We can save a couple characters here without loss of clarity...

```js
{
  "ucan/invoke": ["Qm12345"]
  // ...
}
```

...though I agree that this does look more unusual